src/server/vnparse.c

```diff
@@ -2931,11 +3262,102 @@ add_to_vnl(vnl_t **vnlp, char *noden, char *keyw, char *keyval)
 		free(msgbuf);
 		return (1);
 	}
-
+#endif
 	return (0);
 }
 
+#if !(defined(PBS_MOM) || defined(PBS_PYTHON))
 /** 
+ * @brief
+ *	Check if other resources in 'have' satisfy the remaining
+ *	 resources from 'need'.
+ *
+ * @param[in]	need - the need value of resc_limit_t type.
+ * @param[in]	have - the have value of resc_limit_t type.
+ *
+ * @return int
+ *	1 - if not satisfied
+ *	0 - otherwise
+ * -1 - error
+ */
+static int
+check_other_res(resc_limit_t *need, resc_limit_t *have)
+{
+	resource *pneed, *phave;
+
+	if ((need == NULL) || (have == NULL))
+		return -1;
+
+	if (!GET_NEXT(need->rl_oth_res))
+		return 0;
+
+	for (pneed = (resource *)GET_NEXT(need->rl_oth_res);
+		pneed;
+		pneed = (resource *)GET_NEXT(pneed->rs_link)) {
+		int matched = 0;
+		for (phave = (resource *)GET_NEXT(have->rl_oth_res);
+			phave;
+			phave = (resource *)GET_NEXT(phave->rs_link)) {
+			if (pneed->rs_defin == phave->rs_defin) {
+				resource_def *prdef = pneed->rs_defin;
+				unsigned int atr_type = prdef->rs_type;
+				if ((atr_type == ATR_TYPE_STR)
+					|| (atr_type == ATR_TYPE_BOOL)) {
+					if (!prdef->rs_comp(&pneed->rs_value, &phave->rs_value)) {
+						matched = 1;
+						break;
+					}
+				} else { /* for atr type long, float and size */
+					if (prdef->rs_comp(&pneed->rs_value, &phave->rs_value) <= 0) {
+						matched = 1;
+						break;
+					}
+				}
+			}
+		}
+		if (!matched)
+			return 1;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief
+ *	appends a chunk spec to sched select and also tries to
+ *	group identical chunk specs by increasing the chunk count
+ *
+ * @param[in/out]	new_schedselect - pointer to new sched select
+ * @param[in]		chunkstr - current chunk spec to append
+ * @param[in/out]	tmp_chunk_spec - buffer to accumulate identical chunk specs
+ * @param[in/out]	tmp_chunk_ct - count of gathered identical chunk specs
+ *
+ * @return none
+ */
+static void
+coagulate_append_sched_sel(char *new_schedselect, char *chunkstr, char *tmp_chunk_spec, int *tmp_chunk_ct)
```

Comment:
That's a strange name for a function. The word coagulate is almost always associated with blood or thickening of a liquid. I'd suggest append_and_group_sched_sel(). It's a little longer but more descriptive.

Reply:
Agree, Addressed in commit https://github.com/PBSPro/pbspro/pull/1452/commits/8f309272ad00f7587c830cad07238cb40598e24a